src/parser/jobs/gnb/modules/Ammo.tsx

```diff
@@ -0,0 +1,149 @@
+import {t, Trans} from '@lingui/macro'
+import Color from 'color'
+import {Fragment} from 'react'
+
+import {ActionLink} from 'components/ui/DbLink'
+import TimeLineChart from 'components/ui/TimeLineChart'
+import ACTIONS from 'data/ACTIONS'
+import JOBS from 'data/JOBS'
+import {CastEvent} from 'fflogs'
+import Module, {dependency, DISPLAY_MODE} from 'parser/core/Module'
+import Checklist, {Requirement, Rule} from 'parser/core/modules/Checklist'
+
+const AMMO_GENERATORS = {
+	[ACTIONS.SOLID_BARREL.id]: 1,
+	[ACTIONS.DEMON_SLAUGHTER.id]: 1,
+	[ACTIONS.BLOODFEST.id]: 2,
+}
+
+const AMMO_SPENDERS = {
+	[ACTIONS.GNASHING_FANG.id]: 1,
+	[ACTIONS.BURST_STRIKE.id]: 1,
+	[ACTIONS.FATED_CIRCLE.id]: 1,
+}
+
+const MAX_AMMO = 2
+
+class AmmoState {
+	time: number | null = null
+	ammo: number | null = null
+}
+
+// TODO: Check for bad Burst Strikes (i.e. when at 1 ammo + Bloodfest is on CD + Gnashing is off CD)
```

Comment:
just a suggestion: might be easier for other devs if analytical TODOs were maintained in a job-level todo file. RDM has a pretty good example.